spec/models/resource_pool_spec.rb

```diff
@@ -117,4 +117,16 @@
       expect(subject.current_tenant).to eq(Tenant.root_tenant)
     end
   end
+
+  describe "all_hosts" do
+    let(:resource_pool) { FactoryBot.create(:resource_pool) }
+    let(:host) { FactoryBot.create(:host) }
+    let(:rel) { FactoryBot.create(:relationship, :resource_type => "Host", :resource_id => host.id) }
+
+    it "returns the relationship host" do
+      host.with_relationship_type("ems_metadata") { host.add_child resource_pool }
```

Comment:
So this isn't actually testing your change, if the parent is a `Host` then it is hitting the `[p]` condition.  Test still passes with the `p.kind_of?(Host) ? [p] : p.all_hosts`